hermes/probe/delete/delete_file.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Delete_file implements the probe operation for deleting a file in a
+// storage system.
+
+// Package delete implements the file deletion operation with a storage system.
+package delete
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"google.golang.org/api/iterator"
+
+	probe "github.com/googleinterns/step224-2020/hermes/probe"
+	m "github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	pb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+// DeleteRandomFile deletes a random file in the target storage system bucket.
+// It then checks that the file has been deleted by trying to get the object.
+// Arguments:
+//	- ctx: pass the context so this probe can be cancelled if needed.
+//	- target: pass the target run information.
+//	- client: pass an initialised storage client for this target system.
+//	- logger: pass the logger associated with the probe calling this function.
+//		-> logger can log to Google Cloud if configured to do so.
+// Returns:
+//	- fileID: returns the ID of the file delete OR a missing file to be created if one is found.
+//	- err:
+//		Status:
+//		- StateJournalInconsistent: the file to be deleted does not exist in Hermes' StateJournal.
+//		- FileMissing: the file to be deleted could not be found in the target bucket.
+//		- BucketMissing: the target bucket on this target system was not found.
+//		- ProbeFailed: there was an error during one of the API calls and the probe failed.
+func DeleteRandomFile(ctx context.Context, target *probe.Target, client stiface.Client, logger *logger.Logger) (int32, error) {
```

Comment:
Do we need a function with 5 parameters, and 2 return values, that's only two lines? Or could we just call this directly?

Reply:
This may be a slight over-complication... I'll remove this function and export `pickFileToDelete()` instead.